### PR TITLE
Added gaussian distribution function to RNG

### PR DESCRIPTION
**Pull request by Chaosus**
Some users might wanted it..

image

You can test this function in:
[TestRandom.zip](https://github.com/godotengine/godot/files/2966239/TestRandom.zip)

### CODE REVIEW

**Comment by Naryosha** [comment]
- I think people would be more familiar with parameter in the order of `(mean, sd)`. I looked into C, numpy, R, matlab and they seem to be in that order.
- It seems that RNG class is just an interface for `core/math/random_pcg.*`. Would it be better to put the method there and do `_FORCE_INLINE_` like other methods?

File: doc/classes/RandomNumberGenerator.xml [hunk 1]
```diff
@@ -28,6 +28,17 @@
...
 		</method>
+		<method name="randfn">
+			<return type="float">
+			</return>
+			<argument index="0" name="deviation" type="float" default="1.0">
+			</argument>
+			<argument index="1" name="mean" type="float" default="0.0">
+			</argument>
+			<description>
+				Generates normally distributed pseudo-random number with a standart [code]deviation[/code] and the specified [code]mean[/code].
```
Comment (Naryosha) [inline doc/classes/RandomNumberGenerator.xml]: `standart` -> `standard`

Comment (bojidar-bg) [reply doc/classes/RandomNumberGenerator.xml]: Also, probably mention that the Box-Muller transform is being used.

**Comment by Chaosus** [comment]
@Naryosha / @bojidar-bg Done. Note - Making it FORCE_INLINE inside random_pcg is not possible due to cross-reference to PCG generator in math_funcs header

**Comment by Chaosus** [comment]
Ah.. however I can include math.h itself instead math_funcs...

**Comment by slapin** [comment]
Thank you so much for this!

**Comment by Chaosus** [comment]
Hmm.. for some reason the function generates exraordinary big amount of 0

image
^^ this is histogram of (-100, 100) range

@toasteater I suspect its as you said bug inside randf which affects uniformity - can you check your solutions to fix it ?

**Comment by reduz** [comment]
if you guys find it useful, then for me it's great

**Comment by Chaosus** [comment]
Yeah, I think its ok - it is worked as normal distribution should work, for example 
image

lets test it for 3.2 !

**Comment by hpvb** [comment]
Cherry-picked for 3.1.1.

**Pull request merged**